packages/datastore/system-test/datastore.js

```diff
@@ -927,5 +927,35 @@ describe('Datastore', function() {
         });
       });
     });
+
+    it('should read in a readOnly transaction', function(done) {
+      var transaction = datastore.transaction({ readOnly: true });
+      var key = datastore.key(['Company', 'Google']);
+
+      transaction.run(function(err) {
+        assert.ifError(err);
+        transaction.get(key, done);
+      });
+    });
+
+    it('should not write in a readOnly transaction', function(done) {
```

Comment:
Note that this test fails, which is mentioned in https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4278 under "Integration Tests":

> - A read-only transaction fails if it attempts to write. This test currently fails against the production endpoint.